File: CSES/Introductory Problems/Trailing-Zeros/trailing-zeros.c
#include <stdio.h>

int solution(const int n) {
  int count = 0;
  int divider = 5;
  while (n >= divider) {
    count += n / divider;
    divider *= 5;
  }
  return count;
}
int main(int argc, char *argv[]) {
  int n;
  scanf("%d", &n);
  printf("%d\n", solution(n));
  return 0;
}

File: CSES/Introductory Problems/Trailing-Zeros/trailing-zeros.rs
use std::io::BufRead;
use std::io::Write;
 
fn solution(n: &i32) -> i32 {
    let mut count = 0;
    let mut divider = 5;
    loop {
        count += n / divider;
        divider *= 5;
        if divider > *n {
            break;
        }
    }
    count
}
 
fn main() {
    let mut reader = std::io::stdin().lock();
    let mut writer = std::io::stdout().lock();
    let mut buf = String::new();
    reader.read_line(&mut buf).unwrap();
    let n: i32 = buf.trim().parse().unwrap();
    let res = solution(&n);
    if writeln!(writer, "{}", res).is_err() {
        println!("{}", res);
    }
    if writer.flush().is_err() {
        println!("{}", res);
    }
